### DISCUSSION TITLE
What is the default push size limit of git-lfs?

### DISCUSSION THREAD

**Discussion by abobakrd** [git-lfs/git-lfs#4577] [Q&A] [answered]
Only info i can find is this: https://docs.github.com/en/github/managing-large-files/versioning-large-files/about-git-large-file-storage

But this seems to refer to the repository size. I am trying to push a commit that contains some files with 200mb size (this files are git-lfs tracked), but the push generates the default github push size limit error on 100mb. 

So what is the default (free github and lfs billing plan) git-lfs push size limit and is it different within LFS billing plans? If so, it would be great if someone could provide clear information on this.

**Comment by bk2204** [comment, marked as answer]
Hey,

There are a couple things here.  First, neither Git nor Git LFS have any intrinsic limits on the amount of data that can be pushed.  You should be able to push any data up to or the largest possible file (2^64-1 bytes)  The limits you're mentioning are specific to GitHub itself.  GitHub has a limit on Git blobs that's 100 MB, and that limit doesn't apply to Git LFS, where the limit, depending on plan, is either 2, 4, or 5 GiB.  However, the reason you're seeing this failure is because your history still has the large Git blob somewhere in it that GitHub is seeing, and so you'd need to remove it from the history.

You can do this with `git lfs migrate import --everything` with an appropriate `--include` option.

**Reply by abobakrd** [reply]
Thanks for clarifying! Removed the git history and now it pushed without problems. 👍
